kaggle_environments/agent.py

```diff
@@ -104,20 +110,20 @@ def build_agent(raw, builtin_agents, environment_name):
         return UrlAgent(raw, environment_name), True
 
     # A path exists and attempt to grab the source (fallback to the original string).
+    raw_agent = raw
     if os.path.exists(raw):
-        raw = read_file(raw, raw)
+        raw_agent = read_file(raw, raw)
 
     # Attempt to execute the last callable or just return the string.
     agent = None
 
     def callable_agent(observation, configuration):
         nonlocal agent
         if agent is None:
-            agent = get_last_callable(raw) or raw
-
+            agent = get_last_callable(raw_agent, path=raw) or raw_agent
+        configuration["__raw_path__"] = raw
```

Comment:
Is this needed?

Reply:
this makes it so that people using our starter kits can run bots by passing a path to their main.py file (for wtv language they choose to write the bot in) and code their bot without having to worry about path shenanigans.

Here's an example in our C++ agent's main.py file: https://github.com/Lux-AI-Challenge/Lux-Design-2021/blob/master/kits/cpp/simple/main.py at line 19 we use it. We need the path so that the current working directory can be set for the running process (makes things easier for competitors if they know things are running in the root bot directory).